roles/entitlement_deploy/tasks/main.yml

```diff
@@ -14,18 +14,34 @@
 - name: Compute the md5sum of the entitlement files (debug)
   command: md5sum '{{ entitlement_rhsm }}' '{{ entitlement_pem }}'
 
+- name: Get cluster control plane topology
+  command: oc get infrastructure cluster -o jsonpath='{.status.controlPlaneTopology}'
+  register: topology
+
+- name: Set master as machine config target role in single node clusters
+  when: topology.stdout == "SingleReplica"
+  set_fact:
+      machine_config_role: master
+
+- name: Set worker as machine config target role in non-single-node clusters
+  when: not topology.stdout == "SingleReplica"
+  set_fact:
+      machine_config_role: worker
```

Comment:
@omertuc could you add a way bypass this?
```
if empty(force_machine_config_role):
  try with the topology
else:
  machine_config_role = force_machine_config_role
```
this `force_machine_config_role` can be set in the toolbox, either now or later when we switch to Python+simpler arg parsing

Reply:
hey @omertuc, it would be nice if you could update this PR with at least this comment addressed,
the rest is nothing important, but this comment would allow entitlement deployment on all-in-one clusters

Reply:
Let me know if the recent push works for you